File: backend/app/models/review.rb
class Review < ApplicationRecord
  belongs_to :product

  validates :body, presence: true
  validates :flavor_rating, presence: true, inclusion: { in: 1..5 }
  validates :texture_rating, presence: true, inclusion: { in: 1..5 }
  validates :hardness_level, presence: true, inclusion: { in: 1..5 }
end

File: frontend/src/types/review.ts
export type Product = {
  id: number;
  name: string;
  source: string;
  external_id: string;
  image_url: string;
  product_url: string;
};

export type Review = {
  id: number;
  product_id: number;
  body: string;
  flavor_rating: number;
  texture_rating: number;
  hardness_level: number;
  created_at: string;
  product: Product;
};